app/services/operations/transfers.rb

```
@@ -0,0 +1,26 @@
+module Operations
+  module Transfers
+    # Transfer operation factory. Creates either a OneToOne/ManyToOne/OneToMany
+    # depending on the length of the "from" and "to" arrays
+    def create(from:, to:, transfer_params:)
+      transfer_klass = if from.length == 1 and to.length == 1
+                         OneToOne
+                       elsif from.length > 1 and to.length == 1
+                         ManyToOne
+                       elsif from.length == 1 and to.length > 1
+                         OneToMany
+                       elsif
+                         # TODO better error message
+                         raise 'Unknown type of transfer'
```

maybe `raise ArgumentError, "Unknown type of transfer"` ?